coderd/util/apiversion/apiversion.go

```diff
@@ -0,0 +1,66 @@
+package apiversion
+
+import (
+	"slices"
+	"strconv"
+	"strings"
+
+	"golang.org/x/xerrors"
+)
+
+func New(maj []int, min int) *APIVersion {
+	v := &APIVersion{
+		supportedMajors: maj,
+		supportedMinor:  min,
+	}
+	return v
+}
+
+type APIVersion struct {
+	supportedMajors []int
+	supportedMinor  int
+}
+
+func (v *APIVersion) Validate(version string) error {
+	if len(v.supportedMajors) == 0 {
+		return xerrors.Errorf("developer error: SupportedMajors is empty")
```

Comment:
do this error checking on creation

Reply:
Modified function signature to make this impossible.